File: CS/GCSync/Form1.cs
using DevExpress.XtraBars.Helpers;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraEditors;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.GoogleCalendar;
using Google.Apis.Auth.OAuth2;
using Google.Apis.Calendar.v3;
using Google.Apis.Calendar.v3.Data;
using Google.Apis.Services;
using Google.Apis.Util.Store;
using System;
using System.Data;
using System.Data.SqlClient;
using System.IO;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace GCSync {
    public partial class Form1 : RibbonForm {
        UserCredential credential;
        CalendarService service;
        CalendarList calendarList;
        string activeCalendarId;
        bool allowEventLoad;
        public Form1() {
            InitializeComponent();
            InitSkinGallery();
            schedulerControl.Start = System.DateTime.Now;
            ricbCalendarList.SelectedIndexChanged += RicbCalendarList_SelectedIndexChanged;
            bbiSynchronize.ItemClick += BbiSynchronize_ItemClick;
            gcSyncComponent.ConflictDetected += GcSyncComponent_ConflictDetected;

        }

        private void GcSyncComponent_ConflictDetected(object sender, ConflictDetectedEventArgs e) {
            XtraMessageBox.Show("Google '" + e.Event.Summary + "' Event conflicts with the Scheduler '" +
                e.Appointment.Subject + "' Appointment." + Environment.NewLine + "Synchronizing by the Google Event.",
                "Conflict detected", MessageBoxButtons.OK);
            //uncomment the following line to sync by Scheduler Appointments instead
            //e.GoogleEventIsValid = false;
        }

        //manually trigger synchronization
        private void BbiSynchronize_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            this.gcSyncComponent.Sync();
        }

        //assign a selected calendar ID to the component's CalendarID property
        private void RicbCalendarList_SelectedIndexChanged(object sender, EventArgs e) {
            ComboBoxEdit edit = (ComboBoxEdit)sender;
            string selectedCalendarSummary = (string)edit.SelectedItem;
            CalendarListEntry selectedCalendar = this.calendarList.Items.FirstOrDefault(x => x.Summary == selectedCalendarSummary);
            this.activeCalendarId = selectedCalendar.Id;
            this.gcSyncComponent.CalendarId = selectedCalendar.Id;
            this.gcSyncComponent.Sync(); 
            UpdateBbiAvailability();
        }


        #region Authorization
        async protected override void OnLoad(EventArgs e) {
            base.OnLoad(e);
            Directory.CreateDirectory(Environment.CurrentDirectory + @"\xml");
            try {
                this.credential = await AuthorizeToGoogle();
                this.service = new CalendarService(new BaseClientService.Initializer() {
                    HttpClientInitializer = this.credential,
                    ApplicationName = "Google Sync Sample"
                });
                this.gcSyncComponent.CalendarService = this.service;
                await UpdateCalendarListUI();
                this.allowEventLoad = true;
                UpdateBbiAvailability();
                this.gcSyncComponent.Storage = schedulerStorage;
            }
            catch (Exception ex) {
                MessageBox.Show(ex.Message);
            }
        }

        async Task<UserCredential> AuthorizeToGoogle() {
            using (FileStream stream = new FileStream("credentials.json", FileMode.Open, FileAccess.Read)) {
                string credPath = Environment.GetFolderPath(
                    Environment.SpecialFolder.Personal);
                credPath = Path.Combine(credPath, ".credentials/GoogleSchedulerSync.json");

                return await GoogleWebAuthorizationBroker.AuthorizeAsync(
                    GoogleClientSecrets.Load(stream).Secrets,
                    new String[] { CalendarService.Scope.Calendar },
                    "user",
                    CancellationToken.None,
                    new FileDataStore(credPath, true));
            }
        }
        #endregion

        //retrieve calendars from a signed user account
        async Task UpdateCalendarListUI() {
            CalendarListResource.ListRequest listRequest = this.service.CalendarList.List();
            this.calendarList = await listRequest.ExecuteAsync();
            this.ricbCalendarList.Items.Clear();
            foreach (CalendarListEntry item in this.calendarList.Items)
                if (!ricbCalendarList.Items.Contains(item.Summary)) {
                    this.ricbCalendarList.Items.Add(item.Summary);
                }
            if (!String.IsNullOrEmpty(this.activeCalendarId)) {
                CalendarListEntry itemToSelect = this.calendarList.Items.FirstOrDefault(x => x.Id == this.activeCalendarId);
                this.gcSyncComponent.CalendarId = this.activeCalendarId;
                if (this.ricbCalendarList.Items.Contains(itemToSelect.Summary)) {
                    this.beiCalendarList.EditValue = itemToSelect.Summary;
                }
                else
                    this.activeCalendarId = String.Empty;
            }
            UpdateBbiAvailability();
        }

        //specifies whether the "Sync" button should be enabled
        void UpdateBbiAvailability() {
            this.bbiSynchronize.Enabled = !String.IsNullOrEmpty(this.activeCalendarId) && this.allowEventLoad;
        }

        void InitSkinGallery() {
            SkinHelper.InitSkinGallery(rgbiSkins, true);
        }
    }
}

File: VB/GCSync/Form1.vb
Imports DevExpress.XtraBars.Helpers
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraEditors
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.GoogleCalendar
Imports Google.Apis.Auth.OAuth2
Imports Google.Apis.Calendar.v3
Imports Google.Apis.Calendar.v3.Data
Imports Google.Apis.Services
Imports Google.Apis.Util.Store
Imports System
Imports System.IO
Imports System.Linq
Imports System.Threading
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace GCSync

    Public Partial Class Form1
        Inherits RibbonForm

        Private credential As UserCredential

        Private service As CalendarService

        Private calendarList As CalendarList

        Private activeCalendarId As String

        Private allowEventLoad As Boolean

        Public Sub New()
            InitializeComponent()
            InitSkinGallery()
            schedulerControl.Start = Date.Now
            AddHandler ricbCalendarList.SelectedIndexChanged, AddressOf RicbCalendarList_SelectedIndexChanged
            AddHandler bbiSynchronize.ItemClick, AddressOf BbiSynchronize_ItemClick
            AddHandler gcSyncComponent.ConflictDetected, AddressOf GcSyncComponent_ConflictDetected
        End Sub

        Private Sub GcSyncComponent_ConflictDetected(ByVal sender As Object, ByVal e As ConflictDetectedEventArgs)
            XtraMessageBox.Show("Google '" & e.Event.Summary & "' Event conflicts with the Scheduler '" & e.Appointment.Subject & "' Appointment." & Environment.NewLine & "Synchronizing by the Google Event.", "Conflict detected", MessageBoxButtons.OK)
        'uncomment the following line to sync by Scheduler Appointments instead
        'e.GoogleEventIsValid = false;
        End Sub

        'manually trigger synchronization
        Private Sub BbiSynchronize_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            gcSyncComponent.Synchronize()
        End Sub

        'assign a selected calendar ID to the component's CalendarID property
        Private Sub RicbCalendarList_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim edit As ComboBoxEdit = CType(sender, ComboBoxEdit)
            Dim selectedCalendarSummary As String = CStr(edit.SelectedItem)
            Dim selectedCalendar As CalendarListEntry = calendarList.Items.FirstOrDefault(Function(x) Equals(x.Summary, selectedCalendarSummary))
            activeCalendarId = selectedCalendar.Id
            gcSyncComponent.CalendarId = selectedCalendar.Id
            gcSyncComponent.Synchronize()
            UpdateBbiAvailability()
        End Sub

#Region "Authorization"
        Async Protected Overrides Sub OnLoad(ByVal e As EventArgs)
            MyBase.OnLoad(e)
            Directory.CreateDirectory(Environment.CurrentDirectory & "\xml")
            Try
                credential = Await AuthorizeToGoogle()
                service = New CalendarService(New BaseClientService.Initializer() With {.HttpClientInitializer = credential, .ApplicationName = "Google Sync Sample"})
                gcSyncComponent.CalendarService = service
                Await UpdateCalendarListUI()
                allowEventLoad = True
                UpdateBbiAvailability()
                gcSyncComponent.Storage = schedulerStorage
            Catch ex As Exception
                MessageBox.Show(ex.Message)
            End Try
        End Sub

        Private Async Function AuthorizeToGoogle() As Task(Of UserCredential)
            Using stream As FileStream = New FileStream("credentials.json", FileMode.Open, FileAccess.Read)
                Dim credPath As String = Environment.GetFolderPath(Environment.SpecialFolder.Personal)
                credPath = Path.Combine(credPath, ".credentials/GoogleSchedulerSync.json")
                Return Await GoogleWebAuthorizationBroker.AuthorizeAsync(GoogleClientSecrets.Load(stream).Secrets, New String() {CalendarService.Scope.Calendar}, "user", CancellationToken.None, New FileDataStore(credPath, True))
            End Using
        End Function

#End Region
        'retrieve calendars from a signed user account
        Private Async Function UpdateCalendarListUI() As Task
            Dim listRequest As CalendarListResource.ListRequest = service.CalendarList.List()
            calendarList = Await listRequest.ExecuteAsync()
            ricbCalendarList.Items.Clear()
            For Each item As CalendarListEntry In calendarList.Items
                If Not ricbCalendarList.Items.Contains(item.Summary) Then
                    ricbCalendarList.Items.Add(item.Summary)
                End If
            Next

            If Not String.IsNullOrEmpty(activeCalendarId) Then
                Dim itemToSelect As CalendarListEntry = calendarList.Items.FirstOrDefault(Function(x) Equals(x.Id, activeCalendarId))
                gcSyncComponent.CalendarId = activeCalendarId
                If ricbCalendarList.Items.Contains(itemToSelect.Summary) Then
                    beiCalendarList.EditValue = itemToSelect.Summary
                Else
                    activeCalendarId = String.Empty
                End If
            End If

            UpdateBbiAvailability()
        End Function

        'specifies whether the "Sync" button should be enabled
        Private Sub UpdateBbiAvailability()
            bbiSynchronize.Enabled = Not String.IsNullOrEmpty(activeCalendarId) AndAlso allowEventLoad
        End Sub

        Private Sub InitSkinGallery()
            Call SkinHelper.InitSkinGallery(rgbiSkins, True)
        End Sub
    End Class
End Namespace
